.github/workflows/release.yml

```diff
@@ -227,7 +227,7 @@ jobs:
           mkdir .cargo
           cat > .cargo/config.toml <<EOF
           [target.${{ matrix.config.target }}]
-          rustflags = ["-C", "target-cpu=${{ matrix.target_cpu }}", "-C", "lto", "-C", "embed-bitcode=yes"]
+          rustflags = ["-C", "target-cpu=${{ matrix.target_cpu }}", "-C", "lto=thin", "-C", "embed-bitcode=yes"]
```

Comment:
Should we use `linker-plugin-lto` as well instead of `embed-bitcode`?

Reply:
The use of plugin-based LTO is not that straightforward because we have some non-Rust code being linked to our Rust code by dependencies. "By the book" approach requires quite a bit of manual work without any obvious performance benefit. 10 seconds on a single run do not really count. Although there's still a room for improvement with `CFLAGS`.

I am leaving the commands and their execution times below for a reference:

* Naive LTO approach (`4m19s`): `RUSTFLAGS='-C lto=thin -C embed-bitcode=yes' cargo build --release --bin jormungandr --manifest-path jormungandr/Cargo.toml --target x86_64-unknown-linux-gnu --no-default-features`
* Plugin-based LTO (`4m29s`): `CC='clang' CFLAGS='-flto=thin -fembed-bitcode' RUSTFLAGS='-Clinker-plugin-lto -Clinker=clang -Clink-arg=-fuse-ld=lld' cargo build --release --bin jormungandr --manifest-path jormungandr/Cargo.toml --target x86_64-unknown-linux-gnu --no-default-features`

Reply:
Hm, actually, the whole point is not the compilation time improvements but making better LTO when building from multiple languages with `clang`. Which is not exactly the case, but we can switch C code build to `clang` easily. Might be interesting to actually measure the final performance improvement on some example. But this should definitely be done as a separate task.

Reply:
I was actually referring to the compilation speedup as suggested by the rustc reference, but your point is indeed interesting.